### DISCUSSION TITLE
Permissions: Operation failed due to object-level permissions violation".

### DISCUSSION THREAD

**Discussion by misch42** [netbox-community/netbox#13068] [General]
I have a user that is allowed to create / add devices.  He also has the right to read sites, roles, types. But when creating a new device netbox tells me: "Operation failed due to object-level permissions violation". How can I check, what specific permission is missing?

**Comment by mtinberg** [comment]
I don't know how you'd tell which specific priv is missing, but I would note that creating a Device also creates Interfaces and other records, I think you'd need access to those as well.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Michael Schwartzkopff ***@***.***>
Sent: Sunday, July 2, 2023 10:33 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Permissions: Operation failed due to object-level permissions violation". (Discussion #13068)


I have a user that is allowed to create / add devices. He also has the right to read sites, roles, types. But when creating a new device netbox tells me: "Operation failed due to object-level permissions violation". How can I check, what specific permission is missing?

**Comment by misch42** [comment]
Thanks. Is there a something like a log that tells you what specific permission is missing?

**Comment by mtinberg** [comment]
I don't know offhand, but you can look at the LOGGING in configuration.py and the docs for how to update the logging config and the source to figure out the identifier to turn the auth logging to debug to see if there is a message with more detail.  sudo journalctl -f -u netbox -u netbox-rqworker should tail and follow the logs emitted from the netbox service and job runner either through stderr or syslog.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Reply by misch42** [reply]
I tried this already, but my selinux kicks in and does not allow logging. I did not have the time to disable selinux or to write the policies. I will check when I have some more time.

**Comment by kkthxbye-code** [comment]
Just in case it helps. I tried it out with the permissions you mentioned, and I had no issues creating a device at all in v3.5.4.

Permissions set:

```
dcim | device role, dcim | device type, dcim | site = View
dcim | device = add, view
```

I even tried with no view permission for device and it still worked.

**Comment by mtinberg** [comment]
Ok.

I haven't had problems with SELinux when using syslog as the logging handler and didn't need to write any custom policy files for Netbox.  On my EL7 system /opt/netbox/shared is directly writable for {application,requests}.log files as well, as etc_t owned by netbox:netbox 0640

Good luck


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by imix99** [comment]
I had the same problem as @misch42. 
This happens when I activate the maintenance mode.
